src/TransformOperationExecutor.ts

```diff
@@ -402,10 +409,13 @@ export class TransformOperationExecutor {
 
     if (!this.options.ignoreDecorators && target) {
       // add all exposed to list of keys
-      let exposedProperties = defaultMetadataStorage.getExposedProperties(target, this.transformationType);
+      let exposedProperties = defaultMetadataStorage.getExposedProperties(
+        target as ClassConstructor<any>,
```

Comment:
We have too many casts... but I agree this code is pretty hard to change. I will keep that in mind for a future review.

Reply:
This code is WIP, most of castings will be removed when PR is ready to review, now I am doing it for faster iterations.

Reply:
Ok